app/controllers/appliance_calculator/application_controller.rb

```diff
@@ -2,11 +2,28 @@
 
 module ApplianceCalculator
   class ApplicationController < ApplicationController
+    layout :choose_layout
+    preserve :"no-js"
+
+    def index
+      redirect_to appliance_calculator_daily_usage_creation_step_path("appliance")
+    end
+
     def clear
       session[:unit_rate] = nil
       session[:usages] = nil
 
       redirect_to appliance_calculator_daily_usage_creation_step_path("appliance")
     end
+
+    def choose_layout
+      return "appliance_calculator" if params["no-js"].nil?
```

Comment:
This is a curiosity question - I'm not sure it would work, but could `present?` be used here? Something like 

```
return "appliance_calculator_no_js" if params["no-js"].present?
"appliance_calculator"
```

Reply:
It would if I were converting the param into a bool (it's always a string) so `"false".present?` evaluates to `true`